src/D2L.CodeStyle.Analysis/MutabilityInspector.cs

```diff
@@ -125,7 +125,7 @@ HashSet<ITypeSymbol> typeStack
 
 				{
 					foreach( ISymbol member in type.GetMembers() ) {
-						if( IsMemberMutableRecursive( member, typeStack ) ) {
+						if( !member.IsStatic && IsMemberMutableRecursive( member, typeStack ) ) {

```

Comment:
I definitely agree that the static itself also needs to be `readonly` -- the analyzer already picks that up, but not for types that we don't own, granted. I could probably push the "ignore statics" down to just before checking the type of the static itself.

In your second example, where `Bar` is mutable, that makes the static field `Default` illegal via the analyzer, but that shouldn't make `Foo` itself bad. Example:

```csharp
public class Foo {
    public static readonly Bar Default = new Bar(); // Bar is mutable, so this is not safe, analyzer will report error
}
public readonly static Foo GoodGuy = new Foo(); // this is safe, even if Foo.Default isn't
```
In this particular case, the problem is in the `GoodGuy` field, the problem is in the `Foo.Default` field. So we should report an error _there_ instead.